rust/protocol/src/identity_key.rs

```diff
@@ -51,20 +68,29 @@ impl From<PublicKey> for IdentityKey {
     }
 }
 
-#[derive(Copy, Clone)]
+/// The private identity of a user.
+///
+/// This cryptographic identity is used to sign pre-keys in [PreKeySignalMessage], to conceal
+/// sealed-sender [crate::sealed_sender::UnidentifiedSenderMessage]s, or to initiate the key
+/// derivation function for a [SessionRecord].
+///
+/// Also see [KeyPair].
+#[derive(Debug, Copy, Clone)]
 pub struct IdentityKeyPair {
     identity_key: IdentityKey,
     private_key: PrivateKey,
 }
 
 impl IdentityKeyPair {
+    /// Create a key pair from a public `identity_key` and a private `private_key`.
```

Comment:
Probably worth commenting (and for that matter, we should be `debug_assert!`ing) that the public key in `identity_key` must match the public key derivable from `private_key`.